File: 프로그래머스/0/120866. 안전지대/안전지대.js
function solution(board) {
    const n = board.length;
    const danger = Array.from({ length: n }, () => Array(n).fill(0));
    
    for (let r = 0; r < n; r++) {
        for (let c = 0; c < n; c++) {
            if (board[r][c] === 1) {
                for (let dr = -1; dr <= 1; dr++) {
                    for (let dc = -1; dc <= 1; dc++) {
                        const nr = r + dr;
                        const nc = c + dc;

                        if (nr >= 0 && nr < n && nc >= 0 && nc < n) {
                            danger[nr][nc] = 1;
                        }
                    }
                }
            }
        }
    }
    
    return danger.flat().filter(v => v === 0).length;
}

File: 프로그래머스/0/120866. 안전지대/안전지대.py
def solution(board):
    dx = [-1, -1, -1, 0, 0, 1, 1, 1]
    dy = [-1, 0, 1, -1, 1, -1, 0, 1]

    n = len(board)
    danger_zone = [[0] * n for _ in range(n)]
    
    for r in range(n):
        for c in range(n):
            if board[r][c] == 1:
                for i in range(8):
                    nr, nc = r + dx[i], c + dy[i]
                    if 0 <= nr < n and 0 <= nc < n:
                        danger_zone[nr][nc] = 1
                danger_zone[r][c] = 1

    return sum(row.count(0) for row in danger_zone)